OpenSim/Common/DelimFileAdapter.h

```diff
@@ -25,16 +25,53 @@
 #include "FileAdapter.h"
 #include "TimeSeriesTable.h"
 
+#include <fstream>
+#include <regex>
+
 namespace OpenSim {
 
+class IncorrectNumTokens : public Exception {
+public:
+    IncorrectNumTokens(const std::string& file,
+                       size_t line,
+                       const std::string& func,
+                       const std::string& msg) :
+        Exception(file, line, func) {
+
+        addMessage(msg);
+    }
+};
+
+class DataTypeMismatch : public Exception {
+public:
+    DataTypeMismatch(const std::string& file,
+                     size_t line,
+                     const std::string& func,
+                     const std::string& expected,
+                     const std::string& received) :
+        Exception(file, line, func) {
+        std::string msg = "expected = " + expected;
+        msg += " received = " + received;
+
+        addMessage(msg);
+    }
+};
+
 /** DelimFileAdapter is a FileAdapter that reads and writes text files with
 given delimiters. CSVFileAdapter and MOTFileAdapter derive from this class and
 set the delimiters appropriately for the files they parse. The read/write
 functions return/accept a specific type of DataTable referred to as Table in 
 this class.                                                                   
 Header in the file is assumed to end with string "endheader" occupying a full
 line.                                                                         */
+template<typename T>
 class OSIMCOMMON_API DelimFileAdapter : public FileAdapter {
+    static_assert(std::is_same<T, double>::value ||
+                  std::is_same<T, SimTK::Vec3>::value ||
+                  std::is_same<T, SimTK::Vec6>::value ||
+                  std::is_same<T, SimTK::SpatialVec>::value,
+                  "Template argument T must be one of the following types : "
```

Comment:
I am fine with this limitation, but I think it is possibly too strict. I think any `Vec#` so that even `Quaternion` (`Vec4`) and `UnitVec` (Vec3 of norm 1) would be inherently valid. It is understandable that Matrix types are not supported, so perhaps a check that is not of type Matrix or Vector (not fixed sized) would be more adequate? Let me know if there are other reasons to limit to these 4 types specifically (which would still be great!).